Compiler warning in `subsys/logging/backends/log_backend_mqtt.c` when `CONFIG_LOG_BACKEND_MQTT=y`

### Describe the bug

I'm getting a compiler warning when using `CONFIG_LOG_BACKEND_MQTT=y`:

```
<ws-path>/zephyr/subsys/logging/backends/log_backend_mqtt.c:112:20: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
```

### Steps to reproduce

* create a basic project 
    * CMakeLists.txt:
    ```CMake
     cmake_minimum_required(VERSION 3.20.0)

     find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
     project(log_mqtt_warning)

     FILE(GLOB app_sources src/*.c)
     target_sources(app PRIVATE ${app_sources})
    ```
     * main.c:
     ```c
     int main()
     {
          return 0;
     }
     ```
* Add these configs to `prj.conf`:
```make
CONFIG_LOG=y
CONFIG_NETWORKING=y
CONFIG_MQTT_LIB=y
CONFIG_LOG_BACKEND_MQTT=y
```
* `west build -b nucleo_f429zi`

#### Expected behaviour
Build suceeds without any warnings.

#### Impact
Annoying but not much more but can be a problem if you're trying to build with `CONFIG_COMPILER_OPT="-Wall -Werror"`.

### Relevant log output

```shell
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- CMake version: 3.28.1
-- Zephyr version: 4.2.0-rc2 (<ws-path>/zephyr)
-- Found west (found suitable version "1.4.0", minimum required is "0.14.0")
-- Board: nucleo_f429zi, qualifiers: stm32f429xx
-- Found host-tools: zephyr 0.17.1 (/home/<user>/zephyr-sdk-0.17.1)
-- Found toolchain: zephyr 0.17.1 (/home/<user>/zephyr-sdk-0.17.1)
-- Found GnuLd: /home/<user>/zephyr-sdk-0.17.1/arm-zephyr-eabi/arm-zephyr-eabi/bin/ld.bfd (found version "2.38") 
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/<user>/zephyr-sdk-0.17.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Zephyr version: 4.2.0-rc2 (<ws-path>/zephyr), build: v4.2.0-rc2-69-g7a8bd3dbe752
[53/212] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/backends/log_backend_mqtt.c.obj
<ws-path>/zephyr/subsys/logging/backends/log_backend_mqtt.c: In function 'log_backend_mqtt_topic_set':
<ws-path>/zephyr/subsys/logging/backends/log_backend_mqtt.c:112:20: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  112 |         mqtt_topic = topic;
      |                    ^
[212/212] Linking C executable zephyr/zephyr.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:       87576 B         2 MB      4.18%
             RAM:       42656 B       192 KB     21.70%
             CCM:          0 GB        64 KB      0.00%
        IDT_LIST:          0 GB        32 KB      0.00%
Generating files from <ws-path>/zephyr/dev/log_mqtt_error/build/zephyr/zephyr.elf for board: nucleo_f429zi
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

* Debian GNU/Linux 12
* Zephyr SDK 0.17.1
* Zephyr OS build v4.2.0-rc2-69-g7a8bd3dbe752

### Additional Context

_No response_

--- a/subsys/logging/backends/log_backend_mqtt.c
+++ b/subsys/logging/backends/log_backend_mqtt.c
@@ -22,7 +22,7 @@ LOG_MODULE_REGISTER(log_backend_mqtt, CONFIG_LOG_DEFAULT_LEVEL);
 #include <errno.h>
 
 static bool panic;
-static char *mqtt_topic = CONFIG_LOG_BACKEND_MQTT_TOPIC_DEFAULT;
+static const char *mqtt_topic = CONFIG_LOG_BACKEND_MQTT_TOPIC_DEFAULT;
 static uint8_t log_buf[CONFIG_LOG_BACKEND_MQTT_MAX_MSG_SIZE];
 static uint32_t log_format_current = CONFIG_LOG_BACKEND_MQTT_OUTPUT_DEFAULT;
 
@@ -33,7 +33,7 @@ static int log_output_func(uint8_t *data, size_t length, void *output_ctx)
 	struct mqtt_client *client = (struct mqtt_client *)output_ctx;
 	struct mqtt_publish_param param = {0};
 
-	param.message.topic.topic.utf8 = (uint8_t *)mqtt_topic;
+	param.message.topic.topic.utf8 = (const uint8_t *)mqtt_topic;
 	param.message.topic.topic.size = strlen(mqtt_topic);
 	param.message.topic.qos = CONFIG_LOG_BACKEND_MQTT_QOS;
 	param.message.payload.data = data;

